presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -894,13 +895,15 @@ public GroupedExecutionProperties visitJoin(JoinNode node, Void context)
                     return left;
                 case PARTITIONED:
                     if (left.currentNodeCapable && right.currentNodeCapable) {
+                        checkState(left.totalLifespans == right.totalLifespans);
```

Comment:
nit: Add a message in case it ever fails (e.g.: including left and right actual lifespans)